scripts/synapse_port_db

```diff
@@ -459,17 +462,61 @@ class Porter(object):
             sqlite_engine = create_engine(sqlite_config)
             postgres_engine = create_engine(postgres_config)
 
-            self.sqlite_store = Store(sqlite_db_pool, sqlite_engine)
-            self.postgres_store = Store(postgres_db_pool, postgres_engine)
-
-            yield self.postgres_store.execute(postgres_engine.check_database)
-
             # Step 1. Set up databases.
             self.progress.set_state("Preparing SQLite3")
-            self.setup_db(sqlite_config, sqlite_engine)
+            sqlite_conn = self.setup_db(sqlite_config, sqlite_engine)
+
+            hs = MockHomeserver(
+                self.hs_config,
+                sqlite_engine,
+                sqlite_conn,
+                sqlite_db_pool,
+            )
+
+            hs.setup()
+            self.sqlite_store = hs.get_datastore()
+
+            # Check if all background updates are done, abort if not.
+            updates_complete = yield self.sqlite_store.has_completed_background_updates()
+            if not updates_complete:
+                sys.stderr.write(
+                    "Pending background updates exist in the SQLite3 database."
+                    " Please start Synapse again and wait until every update has finished"
+                    " before running this script.\n"
+                )
+                defer.returnValue(None)
 
             self.progress.set_state("Preparing PostgreSQL")
-            self.setup_db(postgres_config, postgres_engine)
+            postgres_conn = self.setup_db(postgres_config, postgres_engine)
+
+            hs = MockHomeserver(
```

Comment:
the shadowing of `hs` here is a bad smell... I think there's a bunch of stuff here which can be moved out into separate functions, which would limit the scope of lots of variables, which is a good thing from the POV of making sure that variables aren't misused.

Reply:
Agreed, I was trying to limit the changes to keep the PR small enough, but in this case I'll do that.

Reply:
Done in e73b15fe0